components/switch/switch-visibility.js

```diff
@@ -32,6 +41,35 @@ class VisibilitySwitch extends LocalizeCoreElement(SwitchMixin(LitElement)) {
 		return html`<d2l-icon icon="tier1:visibility-show"></d2l-icon>`;
 	}
 
+	get text() {
+		if (this.on && this._hasConditions) {
+			return this.localize('components.switch.visibleWithPeriod');
+		}
+		else if (this.on) {
+			return this.localize('components.switch.visible');
+		}
+		else {
+			return this.localize('components.switch.hidden');
+		}
+	}
+
+	_createLabel() {
+		const tooltipHelpClasses = {
+			'conditions-show': this.on && this._hasConditions
```

Comment:
Just wondering, how come it would be better to specify `switch-visibility` there? 
Wouldn't it already be possible to infer that it has to do with just the visibility switch?

Reply:
It's generally been our best practice to preface classes with the component name. It also makes the class more searchable for someone who needs to find it in the future (e.g., perhaps `conditions-show` was used within something else, `switch-visibility-conditions-show` is less likely to have been)

Reply:
I'd be OK if we wanted to start relaxing this practice a bit. It was originally a thing because IE11 and legacy-Edge didn't support shadow DOM, so all these classes were in the light DOM and could collide with each other. Not an issue anymore now that we hard-block them.

Reply:
So is there still enough of a benefit in prefacing classes with the component name? 
As in, are we now more lenient with not doing that, or do we _prefer_ to not preface them?

Reply:
The way I would go with it is if the class is defined in a shareable `whatever-styles.js` that gets imported into your component (e.g. [typography](https://github.com/BrightspaceUI/core/blob/main/components/typography/typography.js), [link](https://github.com/BrightspaceUI/core/blob/main/components/link/link.js)), then it makes a lot of sense to prefix with `d2l-whatever` since we would still need to worry about collisions from multiple things being imported and used in a component.

But if it's just 100% internal to your component (the class & style is defined in your component and used in your component), then the benefits Margaree mentioned about searchability go away.

Reply:
> The way I would go with it is if the class is defined in a shareable whatever-styles.js that gets imported into your component (e.g. [typography](https://github.com/BrightspaceUI/core/blob/main/components/typography/typography.js), [link](https://github.com/BrightspaceUI/core/blob/main/components/link/link.js))

So essentially if we're `export`ing any style classes, then we should prefix. 
Otherwise it's not necessary. 

Going forward though, do we want that to be more strict (i.e. should I remove the prefixing for that `conditions-show` class styling), or do we want to be loose with it (whether you do it or not in a non-shareable case, doesn't matter)?

Reply:
Correct. I'm OK with not having a hard and fast rule here, so whatever you prefer.